discovery.md

```diff
@@ -521,14 +545,184 @@ When encoded in JSON, the response format MUST adhere to the following:
+  Note: in the case of updating an existing Service, the URL MUST NOT be the
+  same as the Service's URL.
+- `403 Forbidden` if this Discovery Endpoint does not support creation of
+  new Services with pre-defined `id` values. This means that the client
+  would need to use the `POST` method above instead.
+  The HTTP Response Body SHOULD contain information as to why the create
+  failed.
+- `406 Not Acceptable` to indicate that the request message is syntactically
+  invalid.
+  The HTTP Response Body SHOULD contain information as to why the create
+  failed.
+- `409 Conflict` to indicate that the request message is syntactically valid,
+  but the value of one or more of the attributes are invalid - for example, the
+  `name` is not unique.
+  The HTTP Response Body SHOULD contain information as to why the create
+  failed.
+
+Other responses are allowed, but not defined by this specification.
+
+If a `202 Accepted` was returned, an HTTP `GET` to the `Location` endpoint
+can be used to determine the status of the original request. This specification
+defines the following responses:
+- `200 OK` indicates that the `PUT` was successful. The HTTP Response Body
+  MAY be empty - thus clients are expected to use the Service's true URL
+  to retrieve the current version of the Service.
+- `204 No Content` to indicate that the `PUT` request is not completed yet.
+  The HTTP Response Body MAY be empty.
+- `406 Not Acceptable` to indicate that the request message is syntactically
+  invalid.
+  The HTTP Response Body SHOULD contain information as to why the create
+  failed.
+- `409 Conflict` to indicate that the original request message is
+  syntactically valid, but the value of one or more of the attributes are
+  invalid - for example, the `name` is not unique.
+  The HTTP Response Body SHOULD contain information as to why the create
+  failed.
+
+Other responses are allowed, but not defined by this specification, however
+they are related to the `GET` itself and not the original `PUT` request.
+
+How long the Discovery Endpoint will support the original `Location` URL is
+an implementation choice, however, it MUST be available immediately upon
+return of the `202 Accepted`. Meaning, a `404 Not Found` MUST NOT be returned
+in response to a `GET` to the `Location` URL during the time the creation
+request is being processed.
+
+##### `DELETE /services/{id}`
+
+This MUST delete the Service at the referenced URL. If the Service can not
+be deleted then an error MUST be generated.
+
+If an error is not generated, the Discovery Endpoint MUST return one of:
+- `200 OK` if the new Service was deleted
+- `202 Accepted` to indicate that the request has been accepted but not
+  processed yet
+
+Other responses are allowed, but not defined by this specification.
+
+In the case of `202 Accepted` the Discovery Endpoint MUST include an
+HTTP `Location` Header in the response pointing to an endpoint that the
+client can use to determine the status of the delete.
```

Comment:
What does this URL look like - what is the endpoint?

Reply:
that's an impl detail - I'm not sure we need to standardize it. Do we?

Reply:
Maybe not. My brain kept wanting to try and map something in this document to what that URL should be though. Would it be worth noting that the URL in this case is an implementation detail outside of the scope of the spec?

Reply:
added